packages/maplibre/lib/map/create-map.ts

```diff
@@ -0,0 +1,173 @@
+import {
+  MapContext,
+  MapContextLayer,
+  removeSearchParams,
+  ViewByZoomAndCenter,
+} from "@geospatial-sdk/core";
+
+import { Map, MapOptions, StyleSpecification } from "maplibre-gl";
+import { FeatureCollection, Geometry } from "geojson";
+import {
+  OgcApiEndpoint,
+  WfsEndpoint,
+  WmsEndpoint,
+} from "@camptocamp/ogc-client";
+import { createDatasetFromGeoJsonLayer } from "../helpers/map.helpers";
+import { Dataset, PartialStyleSpecification } from "../maplibre.models";
+
+const featureCollection: FeatureCollection<Geometry | null> = {
+  type: "FeatureCollection",
+  features: [],
+};
+
+export async function createLayer(
+  layerModel: MapContextLayer,
+): Promise<PartialStyleSpecification> {
+  const { type } = layerModel;
+
+  switch (type) {
+    case "wms": {
+      const sourceId = `${layerModel.name}`;
+      const layerId = `${layerModel.name}`;
+
+      const endpoint = await new WmsEndpoint(layerModel.url).isReady();
+      let url = endpoint.getMapUrl([layerModel.name], {
+        widthPx: 256,
+        heightPx: 256,
+        extent: [0, 0, 0, 0], // will be replaced by maplibre-gl
+        outputFormat: "image/png",
+        crs: "EPSG:3857",
+      });
+      url = removeSearchParams(url, ["bbox"]);
+      url = `${url.toString()}&bbox={bbox-epsg-3857}`;
+
+      const dataset: Dataset = {
+        sources: {
+          [sourceId]: {
+            type: "raster",
+            tiles: [url],
+            tileSize: 256,
+          },
+        },
+        layers: [
+          {
+            id: layerId,
+            type: "raster",
+            source: sourceId,
+            paint: {},
+          },
+        ],
+      };
+      return dataset;
+    }
+    case "wfs": {
+      const entryPoint = await new WfsEndpoint(layerModel.url).isReady();
+      const url = entryPoint.getFeatureUrl(layerModel.featureType, {
+        asJson: true,
+        outputCrs: "EPSG:4326",
+      });
+      const geojson = await fetchGeoJson(url);
```

Comment:
same as for OGC API, let's just give maplibre the url and let it work out errors and whatnot